src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -0,0 +1,286 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Security;
+using System.Security.Principal;
+using System.Threading;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Security.Authentication;
+using System.Security.Authentication.ExtendedProtection;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Net.Security
+{
+    //
+    // The class maintains the state of the authentication process and the security context.
+    // It encapsulates security context and does the real work in authentication and
+    // user data encryption with NEGO SSPI package.
+    //
+    internal static class NegotiateStreamPal
+    {
+        // value should match the Windows sspicli NTE_FAIL value
+        // defined in winerror.h
+        private const int NTE_FAIL = unchecked((int)0x80090020);
+
+        internal static IIdentity GetIdentity(NTAuthentication context)
+        {
+            Debug.Assert(!context.IsServer, "GetIdentity: Server is not supported");
+
+            string name = context.Spn;
+            string protocol = context.ProtocolName;
+
+            return new GenericIdentity(name, protocol);
+
+        }
+
+        internal static string QueryContextAssociatedName(SafeDeleteContext securityContext)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        internal static string QueryContextAuthenticationPackage(SafeDeleteContext securityContext)
+        {
+            return NegotiationInfoClass.Kerberos;
```

Comment:
Is there anything that could/should be asserted before returning this answer?  Even if we don't anticipate a NTLM change to come anytime soon, it seems unfortunate to have this hard-coded value potentially getting in the way for later.